src/diffusers/models/unet_motion_model.py

```diff
@@ -0,0 +1,813 @@
+                motion_num_attention_heads=motion_num_attention_heads,
+                motion_attention_bias=motion_attention_bias,
+                motion_activation_fn=motion_activation_fn,
+                motion_max_seq_length=motion_max_seq_length,
+            )
+            self.up_blocks.append(up_block)
+            prev_output_channel = output_channel
+
+        # out
+        if norm_num_groups is not None:
+            self.conv_norm_out = nn.GroupNorm(
+                num_channels=block_out_channels[0], num_groups=norm_num_groups, eps=norm_eps
+            )
+            self.conv_act = nn.SiLU()
+        else:
+            self.conv_norm_out = None
+            self.conv_act = None
+
+        conv_out_padding = (conv_out_kernel - 1) // 2
+        self.conv_out = nn.Conv2d(
+            block_out_channels[0], out_channels, kernel_size=conv_out_kernel, padding=conv_out_padding
+        )
+
+    @classmethod
+    def from_unet2d(
+        cls,
+        unet: UNet2DConditionModel,
+        motion_adapter: Optional[MotionAdapter] = None,
+        load_weights: bool = True,
+        **kwargs,
+    ):
+        has_motion_adapter = motion_adapter is not None
+
+        # based on https://github.com/guoyww/AnimateDiff/blob/895f3220c06318ea0760131ec70408b466c49333/animatediff/models/unet.py#L459
+        config = unet.config
+        config["_class_name"] = cls.__name__
+
+        down_blocks = []
+        for down_blocks_type in config["down_block_types"]:
+            if "CrossAttn" in down_blocks_type:
+                down_blocks.append("CrossAttnDownBlockMotion")
+            else:
+                down_blocks.append("DownBlockMotion")
+        config["down_block_types"] = down_blocks
+
+        up_blocks = []
+        for down_blocks_type in config["up_block_types"]:
+            if "CrossAttn" in down_blocks_type:
+                up_blocks.append("CrossAttnUpBlockMotion")
+            else:
+                up_blocks.append("UpBlockMotion")
+
+        config["up_block_types"] = up_blocks
+
+        if has_motion_adapter:
+            config["motion_norm_num_groups"] = motion_adapter.config["motion_norm_num_groups"]
+            config["motion_cross_attention_dim"] = motion_adapter.config["motion_cross_attention_dim"]
+            config["motion_num_attention_heads"] = motion_adapter.config["motion_num_attention_heads"]
+            config["motion_attention_bias"] = motion_adapter.config["motion_attention_bias"]
+            config["motion_activation_fn"] = motion_adapter.config["motion_activation_fn"]
+            config["motion_max_seq_length"] = motion_adapter.config["motion_max_seq_length"]
+            config["motion_layers_per_block"] = motion_adapter.config["motion_layers_per_block"]
+            config["motion_mid_block_layers_per_block"] = motion_adapter.config["motion_mid_block_layers_per_block"]
+
+        model = cls.from_config(config)
+
+        if load_weights:
```

Comment:
I think this should always be True and not exist as a function argument. Why would one **not** load the unet parameters?

Reply:
Hmm it is unlikely. I set it to be an argument in order to enable training a full model (UNet + Motion Modules). It follows the ControlNet design
https://github.com/huggingface/diffusers/blob/c7617e482a522173ea6f922223aa010058552af8/src/diffusers/models/controlnet.py#L425